### ISSUE TITLE
[SearchBox] WCAG2AA - searchBox embedded component button error don't have a discernible text

### ISSUE THREAD

**Issue by JoaoEnesGoncalves** [lumada-design/hv-uikit-react#783]
**Describe the bug**
searchBox embedded component button error don't have a discernible text

**To Reproduce**
1. Run
```
node C:\Git\hv-uikit-react\automation\robot\storybook\_resources/accessibility.js --url=http://localhost:9002/iframe.html?id=coresearchbox--simplesearchbox --standard=WCAG2AA --element="#root>div.Component-content-4" --actions="set field input to show list suggestions"
```

**Screenshots**
```
{
  documentTitle: 'Storybook',
  pageUrl: 'http://localhost:9002/iframe.html?id=coresearchbox--simplesearchbox',
  issues: [
    {
      code: 'button-name',
      type: 'error',
      typeCode: 1,
      message: 'Buttons must have discernible text (https://dequeuniversity.com/rules/axe/3.4/button-name?application=axeAPI)',
      context: '<div class="HvInput-iconClear-166" role="button" tabindex="-1"><div class="HvInput-closeBox-16...</div>',
      selector: '#root > div:nth-child(2) > div > div > div > div:nth-child(2) > div > div',
      runner: 'axe',
      runnerExtras: [Object]
    },
    {
      code: 'WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Div.Name',
      type: 'error',
      typeCode: 1,
      message: 'This element has role of "button" but does not have a name available to an accessibility API. Valid names are: element content, aria-label undefined, aria-labelledby undefined.',
      context: '<div class="HvInput-iconClear-166" role="button" tabindex="-1"><div class="HvInput-closeBox-16...</div>',
      selector: '#root > div:nth-child(2) > div > div > div > div:nth-child(2) > div > div',
      runner: 'htmlcs',
      runnerExtras: {}
    }
  ]
}
```
